testrepository/tests/ui/test_cli.py

```diff
@@ -186,6 +188,30 @@ def test_outputs_tests_to_stdout(self):
                 '...TestCLIUI.test_outputs_tests_to_stdout\n'
                 '...TestCLIUI.test_construct\n', doctest.ELLIPSIS))
 
+    def test_outputs_tests_meta_to_stdout_json(self):
+        ui, cmd = get_test_ui_and_cmd()
+        tests = {
+            'foo': {'profiles': ['1', '2']},
+            'bar': {'profiles': ['3', '4']},
+            }
+        ui.output_tests_meta(tests, 'json')
+        expected_bytes = json.dumps(tests, sort_keys=True).encode('utf8')
```

Comment:
The fact that you need 'sort_keys=True' here hints at connascence of algorithm to me. Instead, how about:

output = json.loads(ui._stdoiut.bufgfer.getValue())
self.assertDictEquals(output, tests)

Then again, if part of the intent is to show that test ids are ordered, then this is probably fine.